File: Roman-to-Integers/quarta.cpp
#include<stdio.h>
#include<iostream>
#include<unordered_map>

using std::string;

std::unordered_map<char, int> romanos = {
    {'I', 1},
    {'V', 5},
    {'X', 10},
    {'L', 50},
    {'C', 100},
    {'D', 500},
    {'M', 1000},
};

int romanToInt(string s){
    int last_one = 1000;
    int result = 0;
    int valor;
    
    for(char c : s){
        valor = romanos.at(c);
        if(valor <= last_one){
            last_one = valor;
            result += valor;
        }else{
            result -= 2*last_one;
            result += valor;
        }
    }
    return result;
}

int main(){
    printf("%d\n", romanToInt("III"));
    printf("%d\n", romanToInt("LXXX"));
    printf("%d\n", romanToInt("XL"));
    printf("%d\n", romanToInt("MCMLXXXIV"));
    return 0;
}

File: Roman-to-Integers/quarta.py
romanos = {
    "I": 1,
    "V": 5,
    "X": 10,
    "L": 50,
    "C": 100,
    "D": 500,
    "M": 1000
}

def romanToInt(s:str) -> int:
    last_one = 1000
    result = 0
    for c in s:
        value = romanos[c]
        if value <= last_one:
            last_one = value
            result += value
        else:
            result -= last_one*2
            result += value
    return result

print(romanToInt("XL"))
